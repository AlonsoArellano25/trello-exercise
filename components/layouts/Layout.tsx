import { Box } from "@mui/material";
import Head from "next/head";
import React, { FC } from "react";
import { Navbar } from "../ui";

interface Props {
  title?: string;
  children?: React.ReactNode;
}

export const Layout: FC<Props> = ({ title = "OpenJira", children }) => {
  return (
    <Box data-testid="main-box" sx={{ flexGrow: 1 }}>
      <Head>
        <title>{title}</title>
      </Head>

      <Navbar />

      <Box data-testid="box" sx={{ padding: "10px 20px" }}>
        {children}
      </Box>
    </Box>
  );
};
