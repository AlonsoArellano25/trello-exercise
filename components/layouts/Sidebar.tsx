import {
  Drawer,
  Box,
  Typography,
  List,
  ListItem,
  ListItemText,
  ListItemIcon,
  Divider
} from "@mui/material";
import React, { useContext } from "react";
import MailOutlineOutlinedIcon from "@mui/icons-material/MailOutlineOutlined";
import InboxOutlinedIcon from "@mui/icons-material/InboxOutlined";
import { UIContext } from "../../context/ui/UIContext";

const menuItems: string[] = ["Inbox", "Starred", "Send Email", "Drafts"];

export const Sidebar = () => {
  const { sidemenuOpen, closeSideMenu } = useContext(UIContext);

  return (
    <Drawer anchor="left" open={sidemenuOpen} onClose={closeSideMenu}>
      <Box sx={{ width: 250 }}>
        <Box sx={{ padding: "5px 10px" }}>
          <Typography variant="h4">Men√∫</Typography>
        </Box>
        <List>
          {menuItems.map((text, index) => (
            <ListItem button key={text}>
              <ListItemIcon>
                {index % 2 ? (
                  <InboxOutlinedIcon />
                ) : (
                  <MailOutlineOutlinedIcon />
                )}
              </ListItemIcon>
              <ListItemText primary={text} />
            </ListItem>
          ))}
        </List>

        <Divider />

        <List data-testid="list-menu">
          {menuItems.map((text, index) => (
            <ListItem button key={text}>
              <ListItemIcon>
                {index % 2 ? (
                  <InboxOutlinedIcon />
                ) : (
                  <MailOutlineOutlinedIcon />
                )}
              </ListItemIcon>
              <ListItemText primary={text} />
            </ListItem>
          ))}
        </List>
      </Box>
    </Drawer>
  );
};
