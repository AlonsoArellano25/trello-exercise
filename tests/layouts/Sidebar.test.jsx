import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import { Sidebar } from "../../components/layouts";

describe("Tests in  <Sidebar />", () => {
  // it("should render properly as h4", () => {
  //   const { container } = render(<Sidebar />);
  //   const h4 = container.querySelector("h4");
  //   expect(h4.innerHTML).toBe("MenÃº");
  // });

  // it("should render properly contain mainBox", () => {
  //     const { getByTestId } = render(<Sidebar />);
  //     expect(getByTestId("list-menu")).toBeTruthy();
  //   });

  it("should match with snapshot", () => {
    const { container } = render(<Sidebar />);
    expect(container).toMatchSnapshot();
  });
});
