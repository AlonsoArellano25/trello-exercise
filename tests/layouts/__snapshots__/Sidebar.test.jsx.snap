// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in  <Sidebar /> should match with snapshot 1`] = `<div />`;
